yt_idv/scene_components/isolayers.py

```diff
@@ -0,0 +1,231 @@
+import numpy as np
+import traitlets
+from OpenGL import GL
+
+from yt_idv.gui_support import add_popup_help
+from yt_idv.opengl_support import TransferFunctionTexture
+from yt_idv.scene_components.base_component import SceneComponent
+from yt_idv.scene_data.block_collection import BlockCollection
+
+
+class Isolayers(SceneComponent):
+    name = "block_isocontours"
+    render_method = "isocontours"
+
+    data = traitlets.Instance(BlockCollection)
+    iso_tol_is_pct = traitlets.Bool(True)  # if True, the tolerance is a fraction
+    iso_log = traitlets.Bool(True)  # if True, iso values are base 10 exponents
+    iso_tolerance = traitlets.List()  # the tolerance for finding isocontours
+    iso_layers = traitlets.List()  # the target values for isocontours
+    iso_layers_alpha = traitlets.List()  # the transparency of isocontours
+    box_width = traitlets.CFloat(0.1)
+    sample_factor = traitlets.CFloat(1.0)
+    transfer_function = traitlets.Instance(TransferFunctionTexture)
+    tf_min = traitlets.CFloat(0.0)
+    tf_max = traitlets.CFloat(1.0)
+    tf_log = traitlets.Bool(True)
```

Comment:
not positive that all these are needed -- copied over from the block component, need to check whether they're needed by the shader.